profiler/src/ProfilerEngine/Datadog.Profiler.Native/ContentionProvider.cpp

```diff
@@ -156,8 +156,11 @@ void ContentionProvider::AddContentionSample(uint64_t timestamp, uint32_t thread
         // We know that we don't have any span ID nor end point details
 
         rawSample.Timestamp = timestamp;
-        auto end_stack = stack.begin() + std::min(stack.size(), static_cast<std::size_t>(rawSample.Stack.Capacity()));
-        std::copy(stack.begin(), end_stack, rawSample.Stack.begin());
+        auto cs = _callstackProvider.Get();
+        auto end_stack = stack.begin() + std::min(stack.size(), static_cast<std::size_t>(cs.Capacity()));
+        std::copy(stack.begin(), end_stack, cs.begin());
+        cs.SetCount(std::min(stack.size(), static_cast<std::size_t>(cs.Capacity())));
+        rawSample.Stack = std::move(cs);
```

Comment:
Maybe we can put the size/nb frames in a variable instead
```suggestion
        const auto nbFrames = std::min(stack.size(), static_cast<std::size_t>(cs.Capacity());
        auto end_stack = stack.begin() + nbFrames);
        std::copy(stack.begin(), end_stack, cs.begin());
        cs.SetCount(nbFrames);
        rawSample.Stack = std::move(cs);
```